packages/xstate-store/src/store.ts

```diff
@@ -373,11 +393,35 @@ export function createStoreWithProducer<
   transitions: {
     [K in keyof TEventPayloadMap & string]: (
       context: NoInfer<TContext>,
-      event: { type: K } & TEventPayloadMap[K]
+      event: { type: K } & TEventPayloadMap[K],
+      enqueue: EnqueueObject<TEmitted>
     ) => void;
   }
+): Store<TContext, ExtractEventsFromPayloadMap<TEventPayloadMap>, TEmitted>;
+
+export function createStoreWithProducer<
+  TContext extends StoreContext,
+  TEventPayloadMap extends EventPayloadMap,
+  TEmitted extends EventObject = EventObject
+>(
+  producer: NoInfer<
+    (context: TContext, recipe: (context: TContext) => void) => TContext
+  >,
```

Comment:
this is an implementation signature, nothing can be inferred from it anyway
```suggestion
  producer: (context: TContext, recipe: (context: TContext) => void) => TContext,
```